lib/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -21,6 +22,60 @@ const meta = {
 	fixable: true,
 };
 
+/** @type {(node: import('css-tree').CssNode) => node is import('css-tree').Value} */
+const isValueNode = (node) => {
+	return node.type === 'Value';
+};
+
+/** @type {(node1: import('css-tree').CssNode, node2: import('css-tree').CssNode) => boolean} */
+const isEqualValueChildNode = (node1, node2) => {
+	if (node1.type !== node2.type) {
+		return false;
+	}
+
+	const node1Unit = 'unit' in node1 ? node1.unit : '';
+	const node2Unit = 'unit' in node2 ? node2.unit : '';
+
+	return node1Unit === node2Unit;
+};
+
+/** @type {(value1: string, value2: string) => boolean} */
+const isEqualValueSyntaxes = (value1, value2) => {
+	if (value1 === value2) {
+		return true;
+	}
+
+	/** @type {import('css-tree').CssNode} */
+	const value1Node = parse(value1, { context: 'value' });
+	const value2Node = parse(value2, { context: 'value' });
+
+	if (
+		!isValueNode(value1Node) ||
+		!isValueNode(value2Node) ||
+		value1Node.children.size !== value2Node.children.size
+	) {
+		return false;
+	}
+
+	const value1Nodes = value1Node.children.toArray();
+	const value2Nodes = value2Node.children.toArray();
+
+	for (let i = 0; i < value1Nodes.length; i++) {
+		const node1 = value1Nodes[i];
+		const node2 = value2Nodes[i];
+
+		if (
+			typeof node1 === 'undefined' ||
+			typeof node2 === 'undefined' ||
+			!isEqualValueChildNode(node1, node2)
+		) {
+			return false;
+		}
+	}
+
+	return true;
+};
```

Comment:
Thank you for this change @kimulaco 🙇 

-------

I did a few tests and this doesn't seem to handle functions. Possible because it parses with css tree into a full value AST while the function to detect similar syntaxes doesn't have recursion?

```css
p { width: calc(10px + 2px); width: calc(10rem + 2rem); }
```

-----

Alternatively you could look purely at the token stream :

```js
const { tokenize, TokenType } = require('@csstools/css-tokenizer');

/** @type {(value1: string, value2: string) => boolean} */
const isEqualValueSyntaxes = (value1, value2) => {
	if (value1 === value2) {
		return true;
	}

	const value1Tokens = tokenize({ css: value1 }).filter((x) => x[0] !== TokenType.Whitespace && x[0] !== TokenType.Comment);
	const value2Tokens = tokenize({ css: value2 }).filter((x) => x[0] !== TokenType.Whitespace && x[0] !== TokenType.Comment);

	if (value1Tokens.length !== value2Tokens.length) return false;

	for (let i = 0; i < value1Tokens.length; i++) {
		const token1 = value1Tokens[i];
		const token2 = value2Tokens[i];

		if (!token1 || !token2) {
			return false;
		}

		if (token1[0] !== token2[0]) {
			// Different token types
			return false;
		}

		if (
			token1[0] === TokenType.Dimension &&
			token2[0] === TokenType.Dimension &&
			token1[4].unit.toLowerCase() !== token2[4].unit.toLowerCase()
		) {
			// Different units
			return false;
		}

		if (
			token1[0] === TokenType.Function &&
			token2[0] === TokenType.Function &&
			token1[4].value.toLowerCase() !== token2[4].value.toLowerCase()
		) {
			// Different functions
			return false;
		}
	}

	return true;
};
```

Reply:
@romainmenke Nice catch!

Reply:
@romainmenke  Thanks! I forgot the function use case.
I will try to fix it to support the function with reference to your suggestion!

Reply:
@romainmenke @ybiquitous Fixed to support function node. https://github.com/stylelint/stylelint/pull/6772/commits/047218e51c6baa146e9aa22e14c3efdca75f341c
When function, check children recursively.

Reply:
Awesome! thank you @kimulaco